time_tracker_api/time_entries/time_entries_repository.py

```diff
@@ -233,6 +235,52 @@ def find_all_v2(
         function_mapper = self.get_mapper_or_dict(mapper)
         return list(map(function_mapper, result))
 
+    def get_last_entry(
+        self,
+        owner_id: str,
+        event_context: EventContext,
+        visible_only=True,
+        mapper: Callable = None,
+    ):
+        query_builder = (
+            CosmosDBQueryBuilder()
+            .add_sql_where_equal_condition({'owner_id': owner_id})
+            .add_sql_order_by_condition('end_date', Order.DESC)
+            .add_sql_limit_condition(1)
+            .add_sql_offset_condition(1)
+            .build()
+        )
+
+        query_str = query_builder.get_query()
+        params = query_builder.get_parameters()
+
+        partition_key_value = self.find_partition_key_value(event_context)
+        result = self.container.query_items(
+            query=query_str,
+            parameters=params,
+            partition_key=partition_key_value,
+        )
+
+        function_mapper = self.get_mapper_or_dict(mapper)
+        return list(map(function_mapper, result))
+
+    def update_last_entry(self, data: dict, event_context: EventContext):
+        owner_id = data.get('owner_id')
+        last_entry = self.get_last_entry(owner_id, event_context)
+        last_entry = last_entry.pop()
+
+        end_date = str_to_datetime(last_entry.end_date)
+        start_date = data.get('start_date')
+        start_date = str_to_datetime(start_date)
+
+        if start_date < end_date:
+            update_date = {'end_date': data.get('start_date')}
+            return self.partial_update(
+                last_entry.id, update_date, event_context
+            )
+
+        return False
```

Comment:
No need to return. But if you want, you can return None explicitly.

Reply:
Yeah, it will be better only does the partial update
With that, the method return None by default